hermes/bin/src/errors.rs

```diff
@@ -35,13 +35,19 @@ impl Errors {
     }
 
     /// Add an error to the `Errors`
-    pub(crate) fn add_err(&mut self, err: anyhow::Error) {
-        self.0.push(err);
+    pub(crate) fn add_err<E>(&mut self, err: E)
+    where E: Into<anyhow::Error> {
+        let err = err.into();
+        match err.downcast::<Errors>() {
+            Ok(errs) => self.0.extend(errs.0),
+            Err(err) => self.0.push(err),
+        }
     }
 
-    /// Merge two `Errors`
-    pub(crate) fn merge(&mut self, other: Self) {
-        self.0.extend(other.0);
+    /// Return a closure that adds an error to the `Errors`
```

Comment:
what's happening here? We are just collecting all the errors?

Reply:
I've added this function which returns a closure to simplify such code from this
```rust
let mut errors = Errors::new();
some_function().unwrap_or_else(|err| errors.add_err(err))
```
to this
```rust
some_function().unwrap_or_else(errors.get_add_err_fn())
```

Reply:
Yea, `add_err` just adds an error into the `Vec`.
The idea of this type is to collect all possible errors into one collection and return a collection of errors, instead of early return on the first error appearance (of course if it is possible to do).
so for example in this lines of codes
```rust

        let mut errors = Errors::new();

        validate_and_write_icon(manifest, &package).unwrap_or_else(errors.get_add_err_fn());
        validate_and_write_metadata(manifest, build_time, package_name, &package)
            .unwrap_or_else(errors.get_add_err_fn());
        write_www_dir(manifest, &package).unwrap_or_else(errors.get_add_err_fn());
        write_share_dir(manifest, &package).unwrap_or_else(errors.get_add_err_fn());
```
we are not returning if ` validate_and_write_icon` fails, we collect error from this function and move forward and only at the end we are returning all collected errors 
```rust
errors.return_result(Self(package))
```